OSDOCS#4896: Adding new parameter for IBM Cloud VPC BYON install use cases

mjpytlak opened the pull request:
Version(s):
4.13+

Issue:
This PR addresses [osdocs-4896](https://issues.redhat.com/browse/OSDOCS-4896). This PR adds the `networkResourceGroupName` installation configuration parameter, which is required for all installation use cases that require an existing VPC (BYON).

Link to docs preview:

- Installing a cluster on IBM Cloud VPC into an existing VPC > [Additional IBM Cloud VPC configuration parameters](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-vpc.html#installation-configuration-parameters-additional-ibm-cloud_installing-ibm-cloud-vpc)
- Installing a cluster on IBM Cloud VPC into an existing VPC > [Sample customized install-config.yaml file for IBM Cloud VPC](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-vpc.html#installation-ibm-cloud-config-yaml_installing-ibm-cloud-vpc)
- Installing a private cluster on IBM Cloud VPC > [Additional IBM Cloud VPC configuration parameters](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-private.html#installation-configuration-parameters-additional-ibm-cloud_installing-ibm-cloud-private)
- Installing a private cluster on IBM Cloud VPC > [Sample customized install-config.yaml file for IBM Cloud VPC](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-private.html#installation-ibm-cloud-config-yaml_installing-ibm-cloud-private)

QE review:
- [ ] QE has approved this change.

jeffnowicki commented:
Instatller Epic for reference: https://issues.redhat.com/browse/CORS-2384

jeffnowicki commented:
@mjpytlak I thought the configuration parameters documentation was a single source doc element that is shared across all install scenarios. Is that true? Or do you 'customize' per install scenario?

They were individually called out (but I'm assuming you'd ideally want it 'single source').
>Installing a cluster on IBM Cloud VPC into an existing VPC > [Additional IBM Cloud VPC configuration parameters](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-vpc.html#installation-configuration-parameters-additional-ibm-cloud_installing-ibm-cloud-vpc)
>Installing a private cluster on IBM Cloud VPC > [Additional IBM Cloud VPC configuration parameters](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-private.html#installation-configuration-parameters-additional-ibm-cloud_installing-ibm-cloud-private)

Here are my initial review comments (which were assuming 'single source'):

1. In the install configuration parameters, there is a section of required parameters. Which might lead custtomer to assume those are the 'only' required parameters. However, additional 'required' parameters appear in later subsections. Suggest providing a mention under Required configuration parameters, that these are the common required parameters. Additional required parameters may need to be specified based on your install scenario.
1. Suggest placing 'Optional Parameters' as the last section and move 'Additional Parameters' ahead of it.
1. Description for `platform.ibmcloud.resourceGroupName`
```
The name of an existing resource group. 

When NOT using an existing VPC for cluster installation, installer provisioned VPC and cluster resources will be placed in this resource group. If not specified, the installer will place those resources into a resource group created by the installer.

When using an existing VPC for cluster installation, the installer provisioned cluster resources will be placed in this resource group. If not specified, the installer will place those resources into a resource group created by the installer. The existing VPC resources must exist in the resource group specified for the `platform.ibmcloud.networkResourceGroupName` parameter.

In both cases, this resource group must only be used for a single cluster installation since the cluster components assume ownership of all of the resources in the resource group.
```

4. Description for `platform.ibmcloud.networkResourceGroupName`
```
Required when using an existing VPC for cluster installation (ignored otherwise). The name of an existing resource group that contains the already created VPC and subnets to be used for cluster installation.
```

jeffnowicki commented:
Similar comment regarding the sample install-config.yaml files. Assumption is that this is a 'single-source' common doc element for all install scenarios. Please confirm @mjpytlak.
>Installing a cluster on IBM Cloud VPC into an existing VPC > [Sample customized install-config.yaml file for IBM Cloud VPC](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-vpc.html#installation-ibm-cloud-config-yaml_installing-ibm-cloud-vpc)
>Installing a private cluster on IBM Cloud VPC > [Sample customized install-config.yaml file for IBM Cloud VPC](https://55612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-private.html#installation-ibm-cloud-config-yaml_installing-ibm-cloud-private)

(I'll be adding review comments soon)

mjpytlak commented:
Hey @jeffnowicki thanks for the feedback. I updated the descriptions. PTAL. Also - answering these in the order your comments:

1. Yes – the installation configuration parameters content is shared across the installation topics for _every_ platform/vendor that OpenShift supports. Having said that, I did condition the content around `resourceGroupName` and `networkResourceGroupName` for IBM scenarios that require an existing VPC. That was probably a bit much, so I removed the condition and now the content will appear in the parameter descriptions for all of the IBM Cloud VPC installation scenarios.
2. I agree with you that we should probably clarify that the parameters listed under `Required configuration parameters` are common to _all_ cloud provider/platforms we support. And that depending on your particular scenario, additional parameters might be required. Given that this content is shared across all of the installation doc, I would prefer to open a separate doc issue to track this update, instead of introducing the update here.
3. For the time being, I would prefer to leave `Optional configuration parameters` section where it is now. I can only speculate, but I believe the intent was to order/group the sections based on common parameters vs. platform-specific parameters, so while they are “optional” they do apply to all supported cloud providers/platforms.
4. Yes. We do share a single sample yaml file example across all of the IBM Cloud VPC install use cases. However, this file contains multiple examples based on the specific installation scenario.  I have only added `resourceGroupName` and `networkResourceGroupName` to the samples for BYON and private (I will add the content to the restricted doc work that is in flight once this PR is approved). Keeping this in mind, I do not believe we have to annotate further.

jeffnowicki commented:
Thank you @mjpytlak.

/lgtm

mjpytlak commented:
@MayXuQQ Ready for QE review. PTAL.

MayXuQQ commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1346,36 +1346,27 @@ Additional IBM Cloud VPC configuration parameters are described in the following
+|The name of an existing resource group.
+By default, an installer-provisioned VPC and cluster resources are placed in this resource group. When not specified, the installation program creates the resource group for the cluster.
+If you are deploying the cluster into an existing VPC, the installer-provisioned cluster resources are placed in this resource group. When not specified, the installation program creates the resource group for the cluster. The VPC resources that you have provisioned must exist in a resource group that you specify using the `networkResourceGroupName` parameter.
+In either case, this resource group must only be used for a single cluster installation, as the cluster components assume ownership of all of the resources in the resource group. [^1^]
```
@mjpytlak  please reference the comments in https://issues.redhat.com/browse/CORS-2557, whether the doc need add more ?

mjpytlak replied:
@MayXuQQ IMO. The existing updates, plus what was written for BYON in 4.12, covers the comments in https://issues.redhat.com/browse/CORS-2557. The one update that I believe is required suggesting that `resourceGroup` should be empty. I will take care of that shortly.

MayXuQQ replied:
@mjpytlak based on my test, on IBM cloud the specified resource group need not be empty, similar to Alibaba cloud, different from Azure

mjpytlak replied:
@MayXuQQ Thank you. In light of this, any additional feedback or is the content good to go?

MayXuQQ commented:
/lgtm

mjpytlak commented:
/cherrypick enterprise-4.13
